examples/kafkaconnect/v1beta1/connector-nokafka.yaml

```
@@ -0,0 +1,273 @@
+# This tests the Connector resource without requiring an actual kafka cluster,
+# which is very slow to create. Creating the connector times out waiting for
+# it to report a "Running" status in aws, but the status is not exposed in
+# the terraform provider, so the subsequent crossplane observe call succeeds.
+# As a result, the test actually passes, and validates that crossplane can
+# Create, Import and Delete this resource.
+apiVersion: kafkaconnect.aws.upbound.io/v1beta1
+kind: Connector
+metadata:
+  annotations:
+    meta.upbound.io/example-id: kafkaconnect/v1beta1/connector
+    uptest.upbound.io/timeout: "2100"
+  labels:
+    testing.upbound.io/example-name: connector-nokafka
+  name: connector-nokafka
+spec:
+  forProvider:
+    capacity:
+      - autoscaling:
+          - maxWorkerCount: 2
+            mcuCount: 1
+            minWorkerCount: 1
+            scaleInPolicy:
+              - cpuUtilizationPercentage: 20
+            scaleOutPolicy:
+              - cpuUtilizationPercentage: 80
+    connectorConfiguration:
+      connector.class: org.apache.kafka.connect.file.FileStreamSinkConnector
+      tasks.max: "1"
+      topics: example
+    kafkaCluster:
+      - apacheKafkaCluster:
+          - bootstrapServers: "localhost:9092"
+            vpc:
+              - securityGroupSelector:
+                  matchLabels:
+                    testing.upbound.io/example-name: connector-nokafka
+                subnetSelector:
+                  matchLabels:
+                    testing.upbound.io/example-name: connector-nokafka
```

We have added two Subnet resources below. I have no idea how it behaves since I haven't tested it, but shouldn't we pass it separately like in the [Cluster.kafka](https://github.com/crossplane-contrib/provider-upjet-aws/blob/0631d73c9ad50555f78989f3f5d098e67e346123/examples/kafka/v1beta1/cluster.yaml#L16) resource?

And as I mentioned above in the reference definition section, I think we should use plural resource names.

```suggestion
                subnetsRefs:
                  - name: subnet-az1
                  - name: subnet-az2
```

---

Because the target field of the reference is an array type, the selector finds multiple matches and happily resolves all of them. 

Here's what the spec looks like once references have been resolved:
```
         vpc:
          - securityGroupRefs:
            - name: sg
            securityGroupSelector:
              matchLabels:
                testing.upbound.io/example-name: connector-nokafka
            securityGroups:
            - sg-040d3c5051ff7055e
            subnetRefs:
            - name: subnet-az1
            - name: subnet-az2
            subnetSelector:
              matchLabels:
                testing.upbound.io/example-name: connector-nokafka
            subnets:
            - subnet-0c401af028b9bc05e
            - subnet-0002763a0a9e8db5b
```